lib/python/qmk/cli/generate/autocorrect_data.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python program to make autocorrect_data.h.
+This program reads "autocorrect_dict.txt" and generates a C source file
+"autocorrect_data.h" with a serialized trie embedded as an array. Run this
+program without arguments like
+$ python3 make_autocorrect_data.py
+Or to read from a different typo dict file, pass it as the first argument like
+$ python3 make_autocorrect_data.py dict.txt
+Each line of the dict file defines one typo and its correction with the syntax
+"typo -> correction". Blank lines or lines starting with '#' are ignored.
+Example:
+  :thier        -> their
+  fitler        -> filter
+  lenght        -> length
+  ouput         -> output
+  widht         -> width
+See autocorrect_dict_extra.txt for a larger example.
+For full documentation, see
+https://getreuer.info/posts/keyboards/autocorrection
+"""
+
+import sys
+import textwrap
+from typing import Any, Dict, Iterator, List, Tuple
+
+from milc import cli
+
+import qmk.path
+from qmk.keyboard import keyboard_completer, keyboard_folder
+from qmk.keymap import keymap_completer, locate_keymap
+
+KC_A = 4
+KC_SPC = 0x2c
+KC_QUOT = 0x34
+
+TYPO_CHARS = dict([
+    ("'", KC_QUOT),
+    (':', KC_SPC),  # "Word break" character.
+] + [(chr(c), c + KC_A - ord('a')) for c in range(ord('a'), ord('z') + 1)])  # Characters a-z.
```

Comment:
pyformat wants it this way:
```python
] + [(chr(c), c + KC_A - ord('a')) for c in range(ord('a'),
                                                  ord('z') + 1)])  # Characters a-z.
```

But pytest does NOT like it, and generates this error:
```
lib/python/qmk/cli/generate/autocorrect_data.py:51:3: W504 line break after binary operator
lib/python/qmk/cli/generate/autocorrect_data.py:52:19: E126 continuation line over-indented for hanging indent
```

Ugh...

Reply:
Uh oh, sorry. I regret that I wrote the dict expression that way. Maybe something like this passes?

```py
LETTERS = 'abcdefghijklmnopqrstuvwxyz'
TYPO_CHARS = dict([
    ("'", KC_QUOT),
    (':', KC_SPC),  # "Word break" character.
] + [(c, ord(c) + KC_A - ord('a')) for c in LETTERS])  # Characters a-z.
```

Reply:
was able to make it happy, without that.

Reply:
Awesome! Thanks, drashna!

Reply:
No, thank you for the awesome code, and the additional improvements!  And helping to push me to get more familiar with python!